lumicks/pylake/kymotracker/kymotrack.py

```diff
@@ -497,12 +497,21 @@ def _merge_tracks(self, starting_track, starting_node, ending_track, ending_node
         if starting_track not in self._src or ending_track not in self._src:
             raise RuntimeError("Both tracks need to be part of this group to be merged")
 
-        starting_node = int(starting_node) + 1
+        starting_node = int(starting_node)
         ending_node = int(ending_node)
 
+        start_time_idx = starting_track.time_idx[starting_node]
+        end_time_idx = ending_track.time_idx[ending_node]
+        assert start_time_idx != end_time_idx, "Cannot connect two points with the same time index."
+
+        if start_time_idx > end_time_idx:
```

Comment:
I would add a code comment here explaining very briefly why ordering is important.